### DISCUSSION TITLE
`DEFAULT_DOMAIN_KEY_LENGTH` as an environment variable

### DISCUSSION THREAD

**Discussion by vitoyucepi** [acmesh-official/acme.sh#4747] [Q&A]
## Question
Is it possible to specify `DEFAULT_DOMAIN_KEY_LENGTH` as an environment variable or in `account.conf`?

## Details
I'm trying to use the command `acme.sh --issue -d domain.com`.
I want to use `rsa2048` as a default key algorithm, but it seems impossible without the explicit command line argument `-k 2048`.

For the first time, `keylength` is set here
https://github.com/acmesh-official/acme.sh/blob/0da839cce35f4ab014a6d62133fac03c8f4c6979/acme.sh#L7217
I think I can't override the variable because `DEFAULT_DOMAIN_KEY_LENGTH` at the moment is set here
https://github.com/acmesh-official/acme.sh/blob/0da839cce35f4ab014a6d62133fac03c8f4c6979/acme.sh#L57
Then it's possible to override it using the command line argument here
https://github.com/acmesh-official/acme.sh/blob/0da839cce35f4ab014a6d62133fac03c8f4c6979/acme.sh#L7504-L7510
After that, it goes straight into the `issue` command here
https://github.com/acmesh-official/acme.sh/blob/0da839cce35f4ab014a6d62133fac03c8f4c6979/acme.sh#L7856-L7858
And to the `createDomainKey` command
https://github.com/acmesh-official/acme.sh/blob/0da839cce35f4ab014a6d62133fac03c8f4c6979/acme.sh#L4488

I think it could be overridden here by setting `_cdl` var in `account.conf`
https://github.com/acmesh-official/acme.sh/blob/0da839cce35f4ab014a6d62133fac03c8f4c6979/acme.sh#L1529-L1536

**Comment by neilpang** [comment]
```
alias  acme.sh="acme.sh -k 2048"
```
